### ISSUE TITLE
Update the documentation for v2.0.0

### ISSUE THREAD

**Issue by WeisLeDocto** [LaboratoireMecaniqueLille/crappy#28]
As quite many things will change in version 2.0.0, the documentation and the tutorials will have to be updated before releasing it.
This is especially true for the tutorials about the creation of new blocks.

**Comment by ForestRupicolous** [comment]
Good idea testing it in a new environment, I still get the same results under Win10:

```
pip freeze
>crappy==2.0.0.dev2
>numpy==1.25.2
```

```
venv\Scripts\python.exe -c "import crappy"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "C:\Tools\python_playground\venv\Lib\site-packages\crappy\__init__.py", line 13, in <module>
    from . import actuator
  File "C:\Tools\python_playground\venv\Lib\site-packages\crappy\actuator\__init__.py", line 14, in <module>    
    from .ft232h import DCMotorHatFT232H
  File "C:\Tools\python_playground\venv\Lib\site-packages\crappy\actuator\ft232h\__init__.py", line 3, in <module>
    from .adafruit_dc_motor_hat import DCMotorHatFT232H
  File "C:\Tools\python_playground\venv\Lib\site-packages\crappy\actuator\ft232h\adafruit_dc_motor_hat.py", line 9, in <module>
    from ...tool.ft232h import FT232HServer as FT232H, USBArgsType
  File "C:\Tools\python_playground\venv\Lib\site-packages\crappy\tool\__init__.py", line 5, in <module>
    from . import ft232h
  File "C:\Tools\python_playground\venv\Lib\site-packages\crappy\tool\ft232h\__init__.py", line 6, in <module>  
    from .usb_server import USBServer, USBArgsType
  File "C:\Tools\python_playground\venv\Lib\site-packages\crappy\tool\ft232h\usb_server.py", line 48, in <module>
    class USBServer(Process):
  File "C:\Tools\python_playground\venv\Lib\site-packages\crappy\tool\ft232h\usb_server.py", line 67, in USBServer
    devices: Dict[str, Device] = dict()
             ~~~~^^^^^^^^^^^^^
  File "C:\Users\m.korinek\AppData\Local\Programs\Python\Python311\Lib\typing.py", line 341, in inner
    return cached(*args, **kwds)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\m.korinek\AppData\Local\Programs\Python\Python311\Lib\typing.py", line 1535, in __getitem__    
    return self.copy_with(params)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\m.korinek\AppData\Local\Programs\Python\Python311\Lib\typing.py", line 1538, in copy_with      
    return _GenericAlias(self.__origin__, params,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\m.korinek\AppData\Local\Programs\Python\Python311\Lib\typing.py", line 1337, in __init__       
    self.__parameters__ = _collect_parameters(args)
                          ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\m.korinek\AppData\Local\Programs\Python\Python311\Lib\typing.py", line 256, in _collect_parameters
    if hasattr(t, '__typing_subst__'):
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Tools\python_playground\venv\Lib\site-packages\crappy\_global.py", line 67, in __getattr__
    raise RuntimeError(f"Missing module: {self._name}\n{self._msg}")
RuntimeError: Missing module: pyusb
The module pyusb is necessary to use this functionality. Please install it and try again !
```

```
python --version
>Python 3.11.1
```

Looks some adafruit / FT232H lib is the reason.

Martin

**Comment by WeisLeDocto** [comment]
I get it now, it has to do with changes in the `typing` module between Python 3.10 and 3.11.
At some point, the `__getattr__()` method of Crappy's OptionalModule is called from the `typing` module. That was not the case in version 3.10. The expected behavior when calling `__getattr__()` in an OptionalModule is to raise an exception, as you experienced.

The faulty section of code was [that line](https://github.com/LaboratoireMecaniqueLille/crappy/blob/735a62e48c9c171ef86b3c0e4e2263daf49b8fd6/src/crappy/tool/ft232h/usb_server.py#L67), where a class attribute was typehinted as an instance of OptionalModule in case `pyusb` is missing. Since it is a class attribute, the line of code is executed at `import` and raises the exception. The attribute should instead be typehinted as a class to be of type `type` (see [this part](https://github.com/python/cpython/blob/9c03215a3ee31462045b2c2ee162bdda30c54572/Lib/typing.py#L264) of the source code of `typing`), hence requiring to change [that other line](https://github.com/LaboratoireMecaniqueLille/crappy/blob/735a62e48c9c171ef86b3c0e4e2263daf49b8fd6/src/crappy/tool/ft232h/usb_server.py#L23) to `Device = OptinalModule`. I'll fix that in version 2.0.0.dev3, and it *should* solve the problem.

Please notice that Crappy version 2.0.0.dev2 is not labeled as compatible with Python 3.11 on PyPI, although we do not prevent installing it on 3.11 in `setup.py`. Adding a test pipeline for various platforms and versions of Python is definitely on my ToDo list. Sorry for the inconvenience, and thank you for bringing this bug up !

Weis

**Comment by WeisLeDocto** [comment]
The documentation is now fully ready for version 2.0.0.
It can be considered that #52 closed this issue.
